Wrapper/WrapperScreen.swift

```diff
@@ -0,0 +1,134 @@
+import UIKit
+
+#if canImport(UIKit)
+
+    /**
+     Embeds a wrapped `Screen`'s backing view controller as a child view controller with a matching frame.
+
+     This `Screen` can be useful wen you'd like to add view controller functionality to an existing `Screen` when you
+     don't have access to the backing view controller. For example, you might use a `WrapperScreen` to present a share
+     sheet over an existing `Screen`.
+
+     `WrapperScreen` is intended for use with a `WrapperScreenViewController` subclass. `WrapperScreenViewController`
+     handles embedding the wrapped `Screen`, adding it's backing view controller as a child, and updating its frame to
+     match the the backing view, forwards child view controller methods, updates the wrapped screen when appropriate.
+
+     Provide a view controller description of your `WrapperScreenViewController` subclass in your conformance to
+     `WrapperScreen`:
+
+     ````
+     struct MyWrapper {
+         var wrapped: Content
+     }
+
+     extension ShareSheetScreen: WrapperScreen, Screen where Content: Screen {
+         public func viewControllerDescription(environment: ViewEnvironment) -> ViewControllerDescription {
+             MyWrapperScreenViewController<Content>.description(for: self, environment: environment)
+         }
+     }
+     ````
+
+     See `WrapperScreenViewController` for more information on how write your view controller subclass.
+     */
+    public protocol WrapperScreen: Screen {
```

Comment:
Naming bikeshed: `ContainerScreen`?

Reply:
Dang I like this name a lot better but I'm worried that `ContainerViewController` overloads this term already and people will assume this is a screen backed by `ContainerViewController` or that they're somehow related.

Maybe it's fine, since each of the types are documented and have clearly different purposes?

Reply:
Yeah I avoided `ContainerScreen` due to `ContainerViewController`—I also think that sharing the same prefix would make it confusing.

I kinda wish `ContainerViewController` was `WorkflowViewController`, but that rename seems like a bit too much of a breaking change.

Open to alternatives!

Reply:
I wonder how painful it would be to rename `ContainerViewController`. I have always disliked the name and agree that `WorkflowViewController` or `WorkflowHostViewController` would be better.

Looking at our repo:

```
git grep "ContainerViewController(workflow:" | wc -l
     180
```

All these seem like they could be simple finds and replaces. We could do it in two phases (rename, deprecated typealias, move over, remove typealias) fairly easily.

Reply:
I'm open to it! Curious how others feel.

Also note this would lead to a 2.0.0 when we do decide to remove the typealiases (which seems totally fine to me!).

Reply:
I'm for this 👍

Reply:
I wonder if that would mean changing https://github.com/square/workflow-kotlin to match anywhere?

Reply:
Well, we've gotten really good at doing such migrations, using `SwiftSyntax`.

Not sure how I feel about the `Workflow` prefix, however `HostViewController` sounds good.